### DISCUSSION TITLE
[Support]: Matrix variables with `.` in them aren't converted to `_` for azure-devops.

### DISCUSSION THREAD

**Discussion by akb825** [github/gh-actions-importer#231] [Feature requests]
### What operating system are you using?

- [X] Linux
- [ ] macOS
- [ ] Windows

### What version of the tool are you using?

gh version 2.20.2 (2022-11-15)
gh actions-importer     github/gh-actions-importer      v1.0
actions-importer/cli    v1.0.14451

### What happened?

When converting pipeline matrices from azure-devops, variables with `.` in them aren't converted to `_`.  For example, a matrix variable `cmake.args` would be converted directly to `matrix.cmake.args` and result in an empty string during evaluation as it's interpreted as an invalid context.

The presence of `.` also confuses variable lookup when combining `env` and `matrix` variables. For example, in one project I had a library variable `Common.ispc.version`, which was converted to `env.ispc_version`. I also had a matrix variable `ispc` as a 0/1 toggle. When converting the jobs, the usage of `$(Common.ispc.version)` was converted to `${{ matrix.ispc }}.version)`, finding the incorrect variable and leaving a trailing `.version)` after the substitution, rather than `${{ env.ispc_version }}`.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Kazmi1100** [comment]
Help wanted

**Comment by Leena8686** [comment]
I'm using Windows

**Comment by mrvishal1143** [comment]
oo

**Comment by Khalid-cloud-tech** [comment]
Based on the information you've provided, it seems that the conversion of pipeline matrices from Azure DevOps to GitHub Actions is causing issues with variable naming and lookup. The presence of periods in variable names is causing confusion with the conversion process, resulting in the incorrect substitution of variables. It may be necessary to modify the variable names to avoid periods or use a different method for variable substitution.

**Comment by noomniyom5** [comment]
good

**Comment by Juancanico5** [comment]
> ### Qué sistema operativo estás usando?
> * [x] linux
> * [ ] Mac OS
> * [ ] ventanas
> 
> ### ¿Qué versión de la herramienta estás usando?
> gh versión 2.20.2 (2022-11-15) gh acciones-importador github/gh-acciones-importador v1.0 acciones-importador/cli v1.0.14451
> 
> ### ¿Qué pasó?
> Al convertir matrices de canalización de azure-devops, las variables que contienen `.`no se convierten a `_`. Por ejemplo, una variable de matriz `cmake.args`se convertiría directamente `matrix.cmake.args`y daría como resultado una cadena vacía durante la evaluación, ya que se interpreta como un contexto no válido.
> 
> La presencia de `.`también confunde la búsqueda de variables al combinar `env`y `matrix`variables. Por ejemplo, en un proyecto tenía una variable de biblioteca `Common.ispc.version`, que se convirtió a `env.ispc_version`. También tenía una variable de matriz `ispc`como un conmutador 0/1. Al convertir los trabajos, el uso de `$(Common.ispc.version)`se convirtió a `${{ matrix.ispc }}.version)`, encontrando la variable incorrecta y dejando un final `.version)`después de la sustitución, en lugar de `${{ env.ispc_version }}`.
> 
> ### Salida de registro relevante
> _Ninguna respuesta_
> 
> ### Código de conducta
> * [x] Acepto seguir el Código de Conducta de este proyecto

**Comment by Juancanico5** [comment]
> > ### Qué sistema operativo estás usando?
> > 
> > * [x]   linux
> > * [ ]   Mac OS
> > * [ ]   ventanas
> > 
> > ### ¿Qué versión de la herramienta estás usando?
> > gh versión 2.20.2 (2022-11-15) gh acciones-importador github/gh-acciones-importador v1.0 acciones-importador/cli v1.0.14451
> > ### ¿Qué pasó?
> > Al convertir matrices de canalización de azure-devops, las variables que contienen `.`no se convierten a `_`. Por ejemplo, una variable de matriz `cmake.args`se convertiría directamente `matrix.cmake.args`y daría como resultado una cadena vacía durante la evaluación, ya que se interpreta como un contexto no válido.
> > La presencia de `.`también confunde la búsqueda de variables al combinar `env`y `matrix`variables. Por ejemplo, en un proyecto tenía una variable de biblioteca `Common.ispc.version`, que se convirtió a `env.ispc_version`. También tenía una variable de matriz `ispc`como un conmutador 0/1. Al convertir los trabajos, el uso de `$(Common.ispc.version)`se convirtió a `${{ matrix.ispc }}.version)`, encontrando la variable incorrecta y dejando un final `.version)`después de la sustitución, en lugar de `${{ env.ispc_version }}`.
> > ### Salida de registro relevante
> > _Ninguna respuesta_
> > ### Código de conducta
> > 
> > * [x]   Acepto seguir el Código de Conducta de este proyecto
> 
> ????

**Comment by Juancanico5** [comment]
> > ### Qué sistema operativo estás usando?
> > 
> > * [x]   linux
> > * [ ]   Mac OS
> > * [ ]   ventanas
> > 
> > ### ¿Qué versión de la herramienta estás usando?
> > gh versión 2.20.2 (2022-11-15) gh acciones-importador github/gh-acciones-importador v1.0 acciones-importador/cli v1.0.14451
> > ### ¿Qué pasó?
> > Al convertir matrices de canalización de azure-devops, las variables que contienen `.`no se convierten a `_`. Por ejemplo, una variable de matriz `cmake.args`se convertiría directamente `matrix.cmake.args`y daría como resultado una cadena vacía durante la evaluación, ya que se interpreta como un contexto no válido.
> > La presencia de `.`también confunde la búsqueda de variables al combinar `env`y `matrix`variables. Por ejemplo, en un proyecto tenía una variable de biblioteca `Common.ispc.version`, que se convirtió a `env.ispc_version`. También tenía una variable de matriz `ispc`como un conmutador 0/1. Al convertir los trabajos, el uso de `$(Common.ispc.version)`se convirtió a `${{ matrix.ispc }}.version)`, encontrando la variable incorrecta y dejando un final `.version)`después de la sustitución, en lugar de `${{ env.ispc_version }}`.
> > ### Salida de registro relevante
> > _Ninguna respuesta_
> > ### Código de conducta
> > 
> > * [x]   Acepto seguir el Código de Conducta de este proyecto
> 
> ????

**Comment by por1981** [comment]
> ### คุณใช้ระบบปฏิบัติการอะไร
> * [x] ลีนุกซ์
> * [ ] แมคโอเอส
> * [ ] หน้าต่าง
> 
> ### คุณใช้เครื่องมือเวอร์ชันใด
> gh เวอร์ชัน 2.20.2 (2022-11-15) gh actions-importer github/gh-actions-importer v1.0 actions-importer/cli v1.0.14451
> 
> ### เกิดอะไรขึ้น
> เมื่อแปลงเมทริกซ์ไปป์ไลน์จาก Azure-devops ตัวแปรที่อยู่`.`ในนั้นจะไม่แปลง`_`เป็น ตัวอย่างเช่น ตัวแปรเมทริกซ์`cmake.args`จะถูกแปลงโดยตรง`matrix.cmake.args`และส่งผลให้เป็นสตริงว่างระหว่างการประเมินเนื่องจากถูกตีความว่าเป็นบริบทที่ไม่ถูกต้อง
> 
> การปรากฏตัวของ`.`ยังสร้างความสับสนในการค้นหาตัวแปรเมื่อทำการรวม`env`และ`matrix`ตัวแปร ตัวอย่างเช่น ในโครงการหนึ่ง ฉันมีไลบรารีตัวแปร`Common.ispc.version`ซึ่งถูกแปลงเป็น`env.ispc_version`. ฉันยังมีตัวแปรเมทริกซ์`ispc`เป็นตัวสลับ 0/1 เมื่อแปลงงาน การใช้งาน of `$(Common.ispc.version)`จะถูกแปลงเป็น`${{ matrix.ispc }}.version)`การค้นหาตัวแปรที่ไม่ถูกต้องและทิ้งการต่อท้ายไว้`.version)`หลังการแทนที่ แทนที่จะ`${{ env.ispc_version }}`เป็น
> 
> ### เอาต์พุตบันทึกที่เกี่ยวข้อง
> _ไม่มีการตอบสนอง_
> 
> ### จรรยาบรรณ
> * [x] ฉันตกลงที่จะปฏิบัติตามแนวทางปฏิบัติของโครงการนี้
